drivers/timer/arcv2_timer0.c

```diff
@@ -351,18 +351,11 @@ u32_t z_timer_cycle_get_32(void)
  */
 void sys_clock_disable(void)
 {
-	unsigned int key;  /* interrupt lock level */
 	u32_t control; /* timer control register value */
 
-	key = irq_lock();
-
-	/* disable interrupt generation */
-
 	control = timer0_control_register_get();
 	timer0_control_register_set(control & ~_ARC_V2_TMR_CTRL_IE);
 
-	irq_unlock(key);
-
 	/* disable interrupt in the interrupt controller */
 
 	irq_disable(IRQ_TIMER0);
```

Comment:
I think feedback of Andy was: why not get rid of this function completely?
If left in, then please state in a comment the assumption that interrupts are disabled when calling.
If left in, maybe move up the irq_disable(IRQ_TIMER0) to avoid the timer interrupt itself causes problems?